Fix pretty-printing for `indirectbr` instructions

RyanGlScott opened the pull request:
Fixes #174.

RyanGlScott committed:
Fix pretty-printing for indirectbr instructions

Fixes #174.

langston-barrett reviewed (approved):
Is there an easy way to add a test for this?

RyanGlScott commented:
OK, _fine_, I'll learn how the `llvm-pretty` test suite works :)

Thankfully, it wasn't that bad! I've added a test.

RyanGlScott commented:
Oh, except that [CI fails](https://github.com/GaloisInc/llvm-pretty/actions/runs/19436905359/job/55610047810?pr=176) on this newly added test:

```
    Stmt 3:                                              FAIL
      test/Output.hs:356:
      MISMATCH between 2l/76c expected and 2l/70c actual
         1| ↱expect> indirectbr␠ptr␠%addr,␠[␠label␠%hello,
          | ↳actual> indirectbr␠ptr␠%addr,␠label␠%hello,
         2| ↱expect> ␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠label␠%world␠]
          | ↳actual> ␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠␠label␠%world
```

What the heck? Somehow, this works locally...

RyanGlScott committed:
ci: Bump `cache-key-prefix`

RyanGlScott commented:
Hm, I wonder if this was due to a GHC recompilation-checking bug of some sort. The [failing CI log](https://github.com/GaloisInc/llvm-pretty/actions/runs/19438735955/job/55616419931#step:15:8) shows the following:

```
+ cabal build -- all
Configuration is affected by the following files:
- cabal.project.local
Resolving dependencies...
Build profile: -w ghc-9.6.7 -O1
In order, the following will be built (use -v for more details):
Warning: -jsem is not supported by the selected compiler, falling back to
normal parallelism control.
 - llvm-pretty-0.13.0.0.101 (test:llvm-pretty-test) (additional components to build)
Preprocessing test suite 'llvm-pretty-test' for llvm-pretty-0.13.0.0.101...
Building test suite 'llvm-pretty-test' for llvm-pretty-0.13.0.0.101...
[4 of 6] Compiling Output [Source file changed]
[6 of 6] Compiling Main [Output changed]
[7 of 7] Linking dist-newstyle/build/x86_64-linux/ghc-9.6.7/llvm-pretty-0.13.0.0.101/t/llvm-pretty-test/build/llvm-pretty-test/llvm-pretty-test [Objects changed]
```

Even though I changed the code in `Text.LLVM.PP` in this PR, that module didn't get recompiled! On the other hand, `Output` (which I also changed in this PR) _did_ get recompiled. Very strange...

In any case, I worked around the issue by bumping the `cache-key-prefix`.

langston-barrett commented:
This is not the first time this repo has hit a caching-related problem. While it may fundamentally be a GHC or Cabal problem, we may want to work around it by just disabling caching for the moment (https://github.com/GaloisInc/.github/issues/64). This certainly isn't optimal, but untrustworthy CI isn't either.
